demosplan/DemosPlanCoreBundle/Event/Procedure/PostProcedureUpdatedEvent.php

```diff
@@ -12,21 +12,66 @@
 
 namespace demosplan\DemosPlanCoreBundle\Event\Procedure;
 
+use DemosEurope\DemosplanAddon\Contracts\Entities\EntityInterface;
 use DemosEurope\DemosplanAddon\Contracts\Events\PostProcedureUpdatedEventInterface;
 use demosplan\DemosPlanCoreBundle\Entity\Procedure\Procedure;
 use demosplan\DemosPlanCoreBundle\Event\DPlanEvent;
+use ReflectionClass;
 
 class PostProcedureUpdatedEvent extends DPlanEvent implements PostProcedureUpdatedEventInterface
 {
-    protected Procedure $procedure;
+    protected Procedure $procedureBeforeUpdate;
+    protected Procedure $procedureAfterUpdate;
 
-    public function __construct(Procedure $procedure)
+    public function __construct(Procedure $procedureBeforeUpdate, Procedure $procedureAfterUpdate)
     {
-        $this->procedure = $procedure;
+        $this->procedureBeforeUpdate = $procedureBeforeUpdate;
+        $this->procedureAfterUpdate = $procedureAfterUpdate;
     }
 
-    public function getProcedure(): Procedure
+    public function getProcedureBeforeUpdate(): Procedure
     {
-        return $this->procedure;
+        return $this->procedureBeforeUpdate;
+    }
+
+    public function getProcedureAfterUpdate(): Procedure
+    {
+        return $this->procedureAfterUpdate;
+    }
+
+    public function getModifiedValues(): array
+    {
+        return $this->determineModifiedValues($this->procedureBeforeUpdate, $this->procedureAfterUpdate);
+    }
+
+    private function determineModifiedValues(EntityInterface $oldEntity, EntityInterface $newEntity): array
+    {
+        $modifiedValues = [];
+
+        $reflectionClass = new ReflectionClass($oldEntity);
+        $properties = $reflectionClass->getProperties();
+
+        foreach ($properties as $property) {
+            $propertyName = $property->getName();
+
+            $oldValue = $property->getValue($oldEntity);
+            $newValue = $property->getValue($newEntity);
+
+            if ($oldValue !== $newValue) {
+                if (is_object($oldValue) && is_object($newValue)) {
+                    $modifiedSubValues = $this->determineModifiedValues($oldValue, $newValue);
+                    if ([] !== $modifiedSubValues) {
+                        $modifiedValues[$propertyName] = $modifiedSubValues;
+                    }
```

Comment:
I'd suggest to somehow keep the information that there are two different instances set in the two entities.

Reply:
How do you mean? For ProcedureSettings::mapHint the result would be: `['settings' => ['mapHint' => ['old' => 'old text', 'new' => 'new Text']]]` so it exist the old and the new value.

Reply:
I reviewed the method more with general usage and reusability in mind. So for example if someone calls this method with procedures like this:
```
$originalProcedure = $procedure;
$procedureClone = clone $procedure;
$settingsClone = clone $procedure->getSettings();
$procedureClone->setSettings($settingsClone);
```
My understanding is that the method would not detect any differences, even though there are different `ProcedureSettings` instances set inside the procedures. This may cause problems at some point. My suggestion is to at least explain the behavior (i.e. no support for different instances if they are otherwise equal) in the docblock of the public method, but leave it at that.